[IMP] Access Rights: Technical Access Rights (#12830)
odoo/documentation: content/applications/general/users/access_rights.rst

   :alt: The Sales apps drop-down menu to set the user's level of permissions.

Manage specific permissions
---------------------------

While access rights are typically managed in a bundle, they can also be set for explicit
permissions. For example, giving a user the :guilabel:`Administrator` permission for **Timesheets**
gives them full access to that app. That user, however, can be made to have full access to the app,
but their ability to manage their own timesheets restricted (e.g., if the payroll administrator is
salaried and does not need to manage their own timesheets).

Manage specific permissions requires :ref:`developer mode <developer-mode>`. After that, navigate to
:menuselection:`Settings app`, and then click :guilabel:`Manage users`, select a user, and go to the
:guilabel:`Technical Access Rights` tab. From here, groups can be edited (which is covered in the
next section) and specific access rights can be managed.

In the :guilabel:`Groups` section, these specific access rights are organized by groups and are
split into sections. If no changes are made to these groups, then their permissions will mirror the
selections made in the :guilabel:`Access Rights` tab.

> - :guilabel:`Selected groups`: access rights set by choices made in the :guilabel:`Access Rights`
>   tab
> - :guilabel:`Groups added automatically`: access rights given as part of the access rights provided
>   by the :guilabel:`Selected groups`

Reviewer: Indeed, these are a bit challenging to define since they are a) a part of a new undocumented feature and b) the tooltips on the UI surrounding the section headings are a bit lame 😄 

Here's what I dug up with the input of SUP and RD team members, who helped piece it together.

 ```suggestion
- :guilabel:`Selected groups`: a list of detailed access rights, set by choices made in the
  :guilabel:`Access Rights` tab.
- :guilabel:`Groups added automatically`: *implied* permissions that are *inherited* with the
  explicit permissions already granted to the user. The values here will match the values listed
  under a given *Group*'s form located under the :menuselection:` Users & Companies --> Groups`
  menu, in the :guilabel:`Inherited` tab.

  .. example::
     When the *Sales Administrator* permission set is assigned to a user, then the *Canned Responses
     Administrator* permissions are inherited automatically. These assignments are reflected across
     the values listed in the :guilabel:`Selected Groups` and :guilabel:`Groups added automatically`
     tables, respectively.
```

I like the deep definition, how the new sections connect to familiar UI/forms in past versions, and the **example** block as supplemental context *before* we get into performing specific, high-impact actions with the rows in the tables. I hope this helps and you find it useful— take what you like, please 🙏

To add a permission to this user profile, click :guilabel:`Add a line` in the :guilabel:`Selected
groups` table, and then add permissions to this user profile. To remove a permission, click the

> :icon:`fa-times` :guilabel:`(cancel)` at the end of that permission's row.

Reviewer: Would mention here that _removing_ permissions may have further permission removal consequences in the **Groups added automatically** table, since one informs the other. Maybe a warning block would be a useful format to emphasize this?

Clicking on the permission itself will open a *group management form*. Learn more about
:ref:`managing groups <access-rights/groups>`.

Any permission in the :guilabel:`Groups added automatically` section are implied or required by the
permission shown in the :guilabel:`Selected groups` section. These cannot be removed, but more users
can be given these permissions by clicking on the permission itself, and then adding the user to
that permission's group.

> .. note::

Reviewer: Good detail! I was wondering what the colors and italics meant!

.. image:: access_rights/user-permissions-dropdown-menu.png
   :alt: The Sales apps drop-down menu to set the user's level of permissions.

Manage specific permissions
---------------------------

While access rights are typically assigned in bundles under specific roles, they can also be set as
explicit permissions.

.. example::
   For example, giving a user the :guilabel:`Administrator` permission for **Timesheets**
   gives them full access to that app. That user, while holding full access, can *still* have their
   ability to manage *their own* timesheets restricted — such as in the case of a salaried payroll
   administrator who does not need to track time.

> To manage specific permissions, :ref:`developer mode <developer-mode>` must be enabled. After that,
> navigate to the :menuselection:`Settings` app. Then click :guilabel:`Manage Users`, select a user,
> and go to the :guilabel:`Technical Access Rights` tab. From here, :guilabel:`Groups` can be edited,
> and specific access rights can be managed across the various sections. If no changes are made to
> these *Groups*, then their permissions will mirror the selections made in the :guilabel:`Access
> Rights` tab.

Reviewer 2: Totally optional, I missed the dev mode step on my first pass! I recommend breaking out this instruction from the paragraph to help it stand out? 
```suggestion
To manage specific permissions, :ref:`developer mode <developer-mode>` must be enabled.

After that, navigate to the :menuselection:`Settings` app. Then click :guilabel:`Manage Users`,
select a user, and go to the :guilabel:`Technical Access Rights` tab. From here, :guilabel:`Groups`
can be edited, and specific access rights can be managed across the various sections. If no changes
are made to these groups, then their permissions will mirror the selections made in the
:guilabel:`Access Rights` tab.
```

- :guilabel:`Selected groups`: a list of detailed access rights, set by choices made in the
  :guilabel:`Access Rights` tab.
- :guilabel:`Groups added automatically`: *implied* permissions that are *inherited* with the
  explicit permissions already granted to the user. The values here will match the values listed

>   under a given *Group*'s form located under the :menuselection:` Users & Companies --> Groups`
>   menu, in the :guilabel:`Inherited` tab.

Reviewer 2: No need to add formatting to "group's", fix menuselection formatting, and add the app to beginning of the menu for clarity of instruction
```suggestion
  under a given group's form located under the :menuselection:`Settings --> Users & Companies -->
  Groups` menu, in the :guilabel:`Inherited` tab.
```

Reviewer: I wrote it this way on purpose so I'd say this suggestion is optional @justinmallette 
- *Group* is referencing a section on the UI / Odoo feature so italicized like this is also fine
- Adding Settings in the menuselection is fine however we're already in the Settings at this point so can just point to the menu that should be present anyway.

.. image:: access_rights/tech-access-rights.png
   :alt: The technical access rights tab opened up for a user profile.

.. example::

>    When the *Sales Administrator* permission set is assigned to a user, then the *Canned Responses
>    Administrator* permissions are inherited automatically. These assignments are reflected across
>    the values listed in the :guilabel:`Selected Groups` and :guilabel:`Groups added automatically`
>    tables, respectively.

Reviewer 2: To be precise with matching the UI, I recommend restating like so with :guilabel:s for the UI text
```suggestion
   When the :guilabel:`Sales` app's :guilabel:`Administrator` permission is assigned to a user, then
   the :guilabel:`Canned Responses Administrator` permissions are inherited automatically. These
   assignments are reflected across the values listed in the :guilabel:`Selected Groups` and
   :guilabel:`Groups added automatically` tables, respectively.
```

To add a permission to this user profile, click :guilabel:`Add a line` in the :guilabel:`Selected
groups` table, and then add permissions to this user profile. To remove a permission, click the
:icon:`fa-times` :guilabel:`(cancel)` at the end of that permission's row.

.. warning::

>    Removing permissions from the *Selected groups* list can impact what permissions are listed in
>    the *Groups added automatically* list, since selected permission groups inform what permission
>    groups are added automatically.

Reviewer 2: UI text should be formatted in guilabel when part of current UI
```suggestion
   Removing permissions from the :guilabel:`Selected Groups` list can impact what permissions are
   listed in the :guilabel:`Groups added automatically` list, since selected permission groups
   inform what permission groups are added automatically.
```

Reviewer: Also optional, it's part of a hypothetical warning not an active instruction

> Clicking on the permission itself will open a *group management form*. Learn more about
> :ref:`managing groups <access-rights/groups>`.

Reviewer 2: No need for italics formatting here
```suggestion
Clicking on the permission itself will open a group management form. Learn more about :ref:`managing
groups <access-rights/groups>`.
```

Any permission in the :guilabel:`Groups added automatically` section are implied or required by the
permission shown in the :guilabel:`Selected groups` section. These cannot be removed, but more users
can be given these permissions by clicking on the permission itself, and then adding the user to
that permission's group.

.. note::

>    Any permission in green is already provided by another permission (for example, setting the
>    **Website** permission to :guilabel:`Editor and Designer` will also give that user the
>    :guilabel:`Restricted Editor` permission). Any permissions in red are conflicting and cannot be
>    active at the same time. Any permissions in *italics* is implied by a :guilabel:`Selected group`
>    (these are usually found in the :guilabel:`Groups added automatically`).

Reviewer 2: Great detail! Again totally optional, but may be visually helpful to format these in a bullet list to appeal to readers skimming through the doc :)
```suggestion
   - Any permission in green is already provided by another permission (for example, setting the
     :guilabel:`Website` app's permission to :guilabel:`Editor and Designer` will also give that
     user the :guilabel:`Restricted Editor` permission).
   - Any permissions in red are conflicting and cannot be active at the same time.
   - Any permissions in *italics* is implied by a :guilabel:`Selected group` (these are usually
     found in the :guilabel:`Groups added automatically`).
```
